great_expectations/rule_based_profiler/domain_builder/column_domain_builder.py

```diff
@@ -196,23 +199,17 @@ def exclude_semantic_types(
     def semantic_type_filter(self) -> Optional[SemanticTypeFilter]:
         return self._semantic_type_filter
 
-    def get_effective_column_names(
+    def get_table_column_names(
         self,
         batch_ids: Optional[List[str]] = None,
         validator: Optional[Validator] = None,
         variables: Optional[ParameterContainer] = None,
     ) -> List[str]:
         """
-        This method applies multiple directives to obtain columns to be included as part of returned "Domain" objects.
+        This method returns all column names available (i.e., prior to any inclusions/exclusions filtering is applied).
         """
-        include_column_names: List[str] = cast(
-            List[str],
-            self._resolve_list_type_property(
-                property_name="include_column_names",
-                property_value_type=list,
-                variables=variables,
-            ),
-        )
+        if self._table_column_names:
+            return self._table_column_names
```

Comment:
Here is the part that prevents the re-calculating the values if we have `_table_column_names` already